R/guide-colorbar.r

```diff
@@ -344,17 +349,48 @@ guide_gengrob.colorbar <- function(guide, theme) {
   hgap <- width_cm(theme$legend.spacing.x  %||% (0.5 * unit(title_fontsize, "pt")))
   vgap <- height_cm(theme$legend.spacing.y %||% (0.5 * unit(title_fontsize, "pt")))
 
-  # label
+  # Labels
+
+  # get the defaults for label justification. The defaults are complicated and depend
+  # on the direction of the legend and on label placement
+  just_defaults <- label_just_defaults.colorbar(guide$direction, label.position)
+  # don't set expressions left-justified
+  if (just_defaults$hjust == 0 && any(is.expression(guide$key$.label))) just_defaults$hjust <- 1
+
+  # get the label theme
   label.theme <- guide$label.theme %||% calc_element("legend.text", theme)
+
+  # We break inheritance for hjust and vjust, because that's more intuitive here; it still allows manual
+  # setting of hjust and vjust if desired. The alternative is to ignore hjust and vjust altogether, which
+  # seems worse
+  if (is.null(guide$label.theme$hjust) && is.null(theme$legend.text$hjust)) label.theme$hjust <- NULL
+  if (is.null(guide$label.theme$vjust) && is.null(theme$legend.text$vjust)) label.theme$vjust <- NULL
+
+  # label.theme in param of guide_legend() > theme$legend.text.align > default
+  hjust <- guide$label.hjust %||% theme$legend.text.align %||% label.theme$hjust %||%
```

Comment:
Is it necessary to have both of `theme$legend.text.align` and `label.theme$hjust`? Shouldn't `label.theme` inherit  from `legend.text`?

Reply:
In the current (released) ggplot2, there are two special theme elements for legend alignment, `legend.text.align` and `legend.title.align`. I think both are obsolete and have issues, for example that they don't apply in a vertical context. I only left them in for backwards compatibility, but I'm happy to take them out. `label.theme` does inherit from `legend.text`.

Did you also have a question about me overriding inheritance of `hjust` and `vjust` for `label.theme`? I'm happy to explain more. It's needed to make guides behave intuitively under default settings, so that, e.g., a horizontal guide with labels underneath has the correct label alignments (`hjust = 0.5` instead of `hjust = 0` and `vjust = 1` instead of `vjust = 0.5`).